backend/app/api/endpoints/knowledge.py

```
@@ -1509,3 +1549,470 @@ def _update_kb_summary_after_deletion(kb_id: int, user_id: int, user_name: str):
+                granted_by_username=granted_by.user_name if granted_by else "Unknown",
+                granted_at=p.granted_at,
+            )
+        )
+
+    return PermissionListResponse(total=len(items), items=items)
+
+
+@permission_router.post("/{kb_id}/permissions")
+async def add_knowledge_permissions(
+    kb_id: int,
+    data: "PermissionCreate",
```

Ruff (0.14.14)

Undefined name `PermissionCreate`

(F821)